desk/app/channels-server.hoon

```diff
@@ -1269,4 +1286,74 @@
       (~(put by waiting.hooks) id.effect [origin +.effect])
     (schedule-waiting +.effect)
   ==
+++  get-hook-template
+  |=  =nest:c
+  ^-  template:h
+  =/  order=(list id-hook:h)  (~(got by order.hooks) nest)
+  =/  crons=(list [id-hook:h job:h])
+    %+  roll  ~(tap by crons.hooks)
+    |=  [[=id-hook:h =cron:h] cr=(list [id-hook:h job:h])]
+    ?~  job=(~(get by cron) nest)  cr
+    (snoc cr [id-hook u.job])
+  =/  ids=(list id-hook:h)  (welp order (turn crons head))
+  =/  hooks=(map id-hook:h hook:h)
+    %-  ~(gas by *(map id-hook:h hook:h))
+    ^-  (list [id-hook:h hook:h])
+    %+  turn  ids
+    |=  =id-hook:h
+    ^-  [id-hook:h hook:h]
+    =/  hook  (~(got by hooks.hooks) id-hook)
+    =/  config  (~(gut by config.hook) nest ~)
+    =/  config-map  (~(put by *(map nest:c config:h)) nest config)
+    :-  id-hook
+    hook(compiled ~, state !>(~), config config-map)
+  :*  nest
+      *data:m
```

Comment:
We don't have template metadata anywhere I don't think. Should it just be removed?

Reply:
yeah I was thinking we'd want it when you can make a static template, but we can save for then